### PR TITLE
[SYCL] Enable standard optimization pipeline for device code

### PR DESCRIPTION
**Pull request by bader**
Add new compiler flag to disable optimizations at compile time:
 -fno-sycl-std-optimizations

 Temporally disable optimizations in some LIT tests.

### CODE REVIEW

**Comment by bader** [comment]
I'd like to commit this patch ASAP to avoid regressions in the future.
There are still a few issues with enabling standard optimization pipeline for SPIR target, but I'll handle them post-commit.
I'm going to introduce a new compiler flag to rollback to old behavior when needed (e.g. -fno-sycl-optimizations).

Let me now if you have any concerns about that plan.

File: clang/test/CodeGenSYCL/inline_asm.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-enable-optimizations -triple spir64-unknown-unknown-sycldevice -emit-llvm -x c++ %s -o - | FileCheck %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -emit-llvm -x c++ %s -o - | FileCheck %s
```
Comment (premanandrao) [inline clang/test/CodeGenSYCL/inline_asm.cpp:1]: Should this line have -fno-sycl-std-optimizations?

Comment (bader) [reply clang/test/CodeGenSYCL/inline_asm.cpp:1]: Original test is compiled with enabled optimizations, so I kept it that way.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -4102,6 +4102,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     if (Args.hasFlag(options::OPT_fsycl_esimd, options::OPT_fno_sycl_esimd,
                      false))
       CmdArgs.push_back("-fsycl-explicit-simd");
+
+    if (!Args.hasFlag(options::OPT_fsycl_std_optimizations,
+                      options::OPT_fno_sycl_std_optimizations, true))
+      CmdArgs.push_back("-fno-sycl-std-optimizations");
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/Clang.cpp:4108]: Could you add a driver test?  Otherwise, LGTM

Comment (bader) [reply clang/lib/Driver/ToolChains/Clang.cpp:4108]: Done.

**Comment by elizabethandrews** [comment]
LGTM except for one question - Should a comment be added to the 2 tests where optimizations were enabled before and now disabled, since this is a temporary change?

**Comment by bader** [comment]
> LGTM except for one question - Should a comment be added to the 2 tests where optimizations were enabled before and now disabled, since this is a temporary change?

I'll open an issue to update all the tests where optimizations were disabled due to different issues (there are more than 2 tests) and add a comment with issue ID to all the tests.

File: sycl/test/basic_tests/accessor/accessor.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fno-sycl-std-optimizations -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (Fznamznon) [inline sycl/test/basic_tests/accessor/accessor.cpp:1]: Are these runtime tests failing with optimizations enabled?

Comment (bader) [reply sycl/test/basic_tests/accessor/accessor.cpp:1]: Yes.

**Pull request merged**